packages/flutter/test/material/bottom_sheet_test.dart

```diff
@@ -1060,6 +1060,62 @@ void main() {
     expect(tester.takeException(), isNull);
   });
 
+  testWidgets('Bottom sheet onRemove LocalHistoryEntry only added for current bottom sheet', (WidgetTester tester) async {
+    // Regression test for https://github.com/flutter/flutter/issues/93717
+    PersistentBottomSheetController<void>? sheetController1;
+    PersistentBottomSheetController<void>? sheetController2;
+    await tester.pumpWidget(MaterialApp(
+      home: Scaffold(
+        body: Builder(builder: (BuildContext context) {
+          return Center(
+            child: Column(
+              children: <Widget>[
+                ElevatedButton(
+                  child: const Text('show 1'),
+                  onPressed: () {
+                    if (sheetController1 != null) {
```

Comment:
The set up is overly complex, something like this may be simpler
```dart
 Scaffold(
        body: Builder(builder: (context) {
          return SafeArea(
            child: Column(
              children: [
                ElevatedButton(
                  child: const Text('show 1'),
                  onPressed: () {

                    sheetController1 =
                        Scaffold.of(context).showBottomSheet<void>(
                              (BuildContext context) => const Text('BottomSheet 1'),
                        );
                  },
                ),
                ElevatedButton(
                  child: const Text('show 2'),
                  onPressed: ()  {
                    sheetController2 =
                        Scaffold.of(context).showBottomSheet<void>(
                              (BuildContext context) => const Text('BottomSheet 2'),
                        );
                  },
                ),
                ElevatedButton(
                  child: const Text('close 1'),
                  onPressed: ()  {
                    sheetController1!.close();
                  },
                ),
              ],
            ),
          );
        }),
      )
      ```
      press show1 -> show 2 -> close 1 will cause the same crash